src/pyOpenMS/addons/MatrixDouble.pyx

```diff
@@ -1,46 +1,32 @@
 from Matrix cimport *
+cimport numpy as np
 
 
 # continue with extra code if needed
+            
 
-    def get_matrix(self):
-        """Cython signature: numpy_matrix get_matrix()
+    def get_matrix_as_view(self):
+        """Cython signature: numpy_matrix get_matrix_as_view()
         """
 
         cdef _Matrix[double] * mat_ = self.inst.get()
 
         cdef unsigned int rows = mat_.rows()
         cdef unsigned int cols = mat_.cols()
+        cdef double* data = mat_.data()
+        # Create a NumPy array from the buffer, specifying not to copy the data.
+        return np.array(<double[:rows * cols:1]> data, order='F', copy=False).reshape(rows, cols, order='F')
```

Comment:
What is your idea behind :rows*cols:1?
Why not shape it directly correctly?
And again, I think you really really should not use reshape.
Pretty sure it needs to make a copy, which explains what you are seeing.

Reply:
As shown in eigency, the right way for views is np stridetricks as_strided

Reply:
Basically just because I found some code for that online. Also seems to work (except that it probably creates a copy ;) )
I think you are right. I once tried the as_strided, but it somehow didn't even find it when included in the pyx file.

Reply:
https://github.com/wouterboomsma/eigency/blob/1b1e61cdf8e78d9f9211e7141989c0fdbff7a3b6/eigency/conversions.pyx#L4

gives me an error about it needs to be defined on the module level.... just dummy user here sorry

Reply:
the pyx are just concatenated to whatever autowrap generates from the pxd.
Therefore all the code in a pyx will be at the class level.

I guess adding it to the top of the pxd instead would solve that.